graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -18,100 +18,92 @@ package com.expediagroup.graphql.generator
+    }
 
-    open fun generate(
+    /**
+     * Generate a schema given a list of objects to parse for the queries, mutations, and subscriptions.
+     */
+    fun generateSchema(
         queries: List<TopLevelObject>,
         mutations: List<TopLevelObject> = emptyList(),
-        subscriptions: List<TopLevelObject> = emptyList(),
-        builder: GraphQLSchema.Builder = GraphQLSchema.newSchema()
+        subscriptions: List<TopLevelObject> = emptyList()
     ): GraphQLSchema {
+        val builder = GraphQLSchema.newSchema()
         builder.query(generateQueries(this, queries))
         builder.mutation(generateMutations(this, mutations))
         builder.subscription(generateSubscriptions(this, subscriptions))
 
         // add unreferenced interface implementations
-        state.additionalTypes.forEach {
+        additionalTypes.forEach {
             builder.additionalType(it)
         }
 
-        builder.additionalDirectives(state.directives.values.toSet())
+        builder.additionalDirectives(directives.values.toSet())
         builder.codeRegistry(codeRegistry.build())
 
-        val schema = config.hooks.willBuildSchema(builder).build()
-
-        // Clean up the classpath scanner
-        subTypeMapper.close()
-
-        return schema
-    }
-
-    open fun graphQLTypeOf(type: KType, inputType: Boolean = false, annotatedAsID: Boolean = false): GraphQLType {
-        val hookGraphQLType = config.hooks.willGenerateGraphQLType(type)
-        val graphQLType = hookGraphQLType
-            ?: generateScalar(this, type, annotatedAsID)
-            ?: objectFromReflection(type, inputType)
-
-        // Do not call the hook on GraphQLTypeReference as we have not generated the type yet
-        val unwrappedType = GraphQLTypeUtil.unwrapType(graphQLType).lastElement()
-        val typeWithNullability = graphQLType.wrapInNonNull(type)
-        if (unwrappedType !is GraphQLTypeReference) {
-            return config.hooks.didGenerateGraphQLType(type, typeWithNullability)
-        }
-
-        return typeWithNullability
+        return config.hooks.willBuildSchema(builder).build()
     }
 
-    private fun objectFromReflection(type: KType, inputType: Boolean): GraphQLType {
-        val cacheKey = TypesCacheKey(type, inputType)
-        val cachedType = state.cache.get(cacheKey)
-
-        if (cachedType != null) {
-            return cachedType
-        }
-
-        val kClass = type.getKClass()
-        val graphQLType = state.cache.buildIfNotUnderConstruction(kClass, inputType) { getGraphQLType(kClass, inputType, type) }
-
-        return config.hooks.willAddGraphQLTypeToSchema(type, graphQLType)
+    /**
+     * Clean up the state and saved information after schema generation.
+     *
+     * Not required to call, as the state and subTypeMapper will be cleaned up by the garbage collector,
+     * but it can help clean up memory early if it not being used.
+     */
+    fun close() {
```

Comment:
i think this should be called internally from the `generateSchema`? do we need to expose it?

Reply:
I can move this to generateSchema